src/vivarium/framework/resource/resource.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from vivarium import Component
+    from vivarium.manager import Manager
+
+
+@dataclass
+class Resource:
+    """A generic resource representing a node in the dependency graph."""
+
+    resource_type: str
+    """The type of the resource."""
+    name: str
+    """The name of the resource."""
+    # TODO [MIC-5452]: all resources should have a component
+    component: Component | Manager | None
+    """The component that creates the resource."""
+
+    @property
+    def resource_id(self) -> str:
+        """The long name of the resource, including the type."""
+        return f"{self.resource_type}.{self.name}"
+
+    @property
+    def is_initialized(self) -> bool:
+        """Return True if the resource needs to be initialized."""
+        return False
+
+
+class NullResource(Resource):
```

Comment:
This is because a component may have an `on_initialize_simulants` method that has required resources, even if that component doesn't create any columns. This is the resource that corresponds to that case.